public/app/plugins/datasource/loki/querybuilder/components/LokiQueryBuilder.tsx

```diff
@@ -81,7 +81,9 @@ export const LokiQueryBuilder = React.memo<Props>(
 
       let values;
       const labelsToConsider = query.labels.filter((x) => x !== forLabel);
-      if (labelsToConsider.length === 0) {
+      // If we have no equality/regex operation, we can't fetch series as it will throw an error, so we fetch label values
+      const hasEqualityOperation = labelsToConsider.find((filter) => filter.op === '=' || filter.op === '=~');
```

Comment:
Problem with `=~` is that it would still fail with something like `foo=~".*"`, basically every matcher that would match an empty value: https://github.com/grafana/loki/blob/main/pkg/util/matchers.go#L16

I think we would need the same logic, if `=~` is selected, we need to match the "value" against an empty string, if that regex is valid, we should call labelValues instead.

Reply:
Updated and added test in https://github.com/grafana/grafana/pull/82251/commits/35bdc551f38d0518f09ec88b755274e204a2ab9e